Updated setup.md

Klaushayan opened the pull request:
Basically #63, but also added cargo-make in the setup section, and did what @Bromeon had suggested on #63. I hope this'd be useful.

Bromeon commented:
Thank you for the update! As a tip, you can also reuse the same pull request by force-pushing to the branch (`git push --force-with-lease`). Titles and descriptions can also be edited after you've created the PR 🙂 

Some feedback:
* Maybe mention that while a lot of stuff is already done in the template, it requires some extra dependencies and toolchains to be set up. No need to go into details, but just make sure the user knows what to expect.
* I would remove the sections _Setup_ and _Usage_ entirely. The user will need to check out the template's documentation already, and we just risk becoming outdated and needing to chase external changes in the book.

Klaushayan commented:
Thank you for the tip, I'm fairly new to PRs as you can probably tell lol

I removed the setup and usage sections, replaced the wiki with the template's wiki, mentioned extra dependencies, also replaced "work-in-progress" with "not maintained by us", it made more sense to me.

Bromeon commented on src/getting-started/setup.md:
```diff
@@ -47,18 +47,8 @@ clang -v
 ## Using the template
 
-We also provide an experimental [template](https://github.com/godot-rust/godot-rust-template) to get you started right away. All the boilerplate stuff is already done for you. If you have decided to use the template, you can skip section 2.2. Check out the [wiki](https://github.com/godot-rust/godot-rust-template/wiki) for instructions.
+> If you have decided to use the template, you can skip section 2.2.
```
I think we can remove this line. It applies to virtually any sections, and numeric section references are at the risk of becoming outdated.

Bromeon reviewed (commented):
One minor comment, otherwise looks good to me! I like the additional changes 🙂

Bromeon commented:
Awesome, thanks a lot!
Your changes are now live: https://godot-rust.github.io/book/getting-started/setup.html#using-the-template
